### DISCUSSION TITLE
Temperature Sensor Add-On

### DISCUSSION THREAD

**Discussion by kurtdwarwick** [Luligu/matterbridge-shelly#59] [Q&A] [answered]
Hi @Luligu 

I have been very happy with being able to have my Shelly devices exposed to matter, so well done and thank you.

There is one device, however, that is lacking the data from the temperature sensor add on attached to it. Would it be feasible to have that be exposed?

Also, would you consider adding the ability to expose devices as different types? 

My exact use case is a Shelly 1 with the temperature probe inserted into a water heater, and is currently exposed to HomeKit as a thermostat via MQTT. 

I’d be keen to hear your thoughts.

Thanks 😊

**Comment by Luligu** [comment]
Hi,
thanks you for your appreciations.

"Also, would you consider adding the ability to expose devices as different types?"
What do you mean by that?

You are right the add-on is not supported yet.
I will add it soon, can you send me please the file in /Matterbridge/matterbridge-shelly that starts with the shelly id of the device you have add-on installed?

**Comment by tammeryousef1006** [comment, marked as answer]
Temperature Sensor Add-On being added already
